libbeat/publisher/pipeline/retry.go

```diff
@@ -36,9 +38,10 @@ type retryer struct {
 
 	consumer *eventConsumer
 
-	sig chan retryerSignal
-	out workQueue
-	in  retryQueue
+	sig        chan retryerSignal
+	out        workQueue
+	in         retryQueue
+	doneWaiter *sync.WaitGroup
```

Comment:
you don't need a pointer. Just use `sync.WaitGroup`. The zero value is a valid instance.